### DISCUSSION TITLE
Can't Use - Nothing happens when clicking Icon (and icon always colored)

### DISCUSSION THREAD

**Discussion by rag-user** [aclap-dev/video-downloadhelper#1254] [Help and Q&A]
### Your question

Hi,

This situation started a few days ago with my firefox browser. VDH icon stays always colored, even if no video on page, and nothing happens (no dropdown menu) when I click the VDH icon. (So I can't download, or access settings or do anything with VDH.) 

I tried to update firefox, delete VDH cookies, clear firefox cache and reinstall VDH, but same situation remains after all those attempts. Also, VDH disappeared from my context menu (right-click on page/video) months ago, so can't use that option either. On the other hand, VDH works and behaves normally on Edge, so problem is only with Firefox. (But Firefox is my main browser and the one for which I bought the VDH license, so I need VDH to work on firefox.) I couldn't find any post with similar problem,.

Any idea what may be the problem and how to resolve it?

Thanks in advance.

### Addon & CoApp version and details

Can't copy addon info because can't get to setting menu. Here is the core info I can find:
VDH version - 9.1.0.44
CoApp - there is one, but don't know the version
Firefox version - 130.0 (also didn't work on 127.0)
Windows version - Windows 10 Pro version 22H2

**Comment by Mattmjs** [comment]
Try removing the extension and installing again and see if that helps or not.

**Reply by rag-user** [reply]
Thanks for your suggestion. I tried it several times before (as I wrote in my post), and it didn't work. But because of your suggestion, I tried it again, but decided to make it even more clear, by deleting VDH, deleting all its cookies (I believe) and the browser cache and restarting Firefox. Because I had several firefox tabs open, which I want to keep, I ended firefox in task manager and restarted it. That caused VDH to magically re-appear and still behave badly. (Perhaps firefox was saved before the VDH removal.) I then removed VDH again, but this time restarted firefox "gently" (not an obvious option because not in settings; did it through about:profiles). This time, VDH did not appear by itself, and after installation appears now to function properly.

I have tried yesterday to also gently restart firefox through about:profiles, and I did re-install VDH yesterday too, so not sure what the difference is from today. But perhaps the answer is to make sure it is re-installed cleanly. 

Thanks again for your suggestion and for making me try again this route. Hopefully, also a helpful issue for the developers.

**Reply by GrampaWildWilly** [reply]
If there's tabs you want to reopen, just make them bookmarks.  Or save the URLs in a text file & reopen them that way.  It's really best to close Firefox completely & just reopen it cold.  There are times when you just have to bite the bullet & give in.

**Reply by rag-user** [reply]
So it seems, but I thought that removing an add-on means just that - removing it - and not that firefox needs to be restarted cold for it to happen. Also, I still don't understand what happened to VDH that caused it to stop reacting. Hopefully just a coincidental fatal error and not a bug that will happen again. In any case, thanks for the input.

**Reply by GrampaWildWilly** [reply]
There is a problem discussed on here about how no matter what, the sidebar was opening in Firefox.  The user even removed VDH & Firefox still started up with a sidebar open.  I searched the web & found some advice to remove a file that nobody would ever reason out or guess.  Of course, you have to close all Firefox windows to delete the file.  A while ago I also removed & reinstalled VDH one time & the reinstall magically retained all my settings.  I thought that's not right.  So I removed VDH again, closed Firefox, cold restarted it, & reinstalled VDH.  That made me set all my VDH settings over again.  So Firefox must be holding stuff ready for a possible reinstall unless you close it completely & do a cold restart.

I suspect that there's some file in the innards of Firefox that got corrupted & that explains your VDH problem.  How did it get corrupted?  Was it Firefox corrupting the file, VDH, or some other villain?  And what action you took was it that caused that file to get reconstructed?  I think the answer is to thank your lucky stars & run to the next fun thing to do.
